lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -320,10 +329,89 @@ fun ExprValue.cast(
                             "and the date format to be YYYY-MM-DD", internal = false, cause = e)
                     }
                 }
+                TIME -> {
+                    val (timezone, precisionProvided) = targetDataType.args.map { it.toInt() }
+                    when {
+                        // Even though the target type is TIME, the value may differ depending on timezone and precision provided.
+                        type == TIME -> {
+                            val time = timeValue()
+                            val precision = when {
+                                // precisionProvided > MAX_PRECISION_FOR_TIME indicates that the precision was not specified.
+                                precisionProvided > MAX_PRECISION_FOR_TIME -> time.precision
+                                else -> precisionProvided
+                            }
+                            val tzMinutes = if (timezone == 1) {
+                                time.zoneOffset?.totalSeconds?.div(SECONDS_PER_MINUTE) ?: LOCAL_TIMEZONE_OFFSET.totalSeconds / SECONDS_PER_MINUTE
+                            } else {
+                                null
+                            }
+                            return valueFactory.newTime(Time.of(time.localTime.hour, time.localTime.minute, time.localTime.second, time.localTime.nano, precision, tzMinutes))
+                        }
+                        type == TIMESTAMP -> {
+                            val ts = timestampValue()
+                            val precision = when {
+                                precisionProvided > MAX_PRECISION_FOR_TIME -> ts.decimalSecond.scale()
+                                else -> precisionProvided
+                            }
+                            return valueFactory.newTime(
+                                ts.hour,
```

Comment:
I think this is handled in a dependent PR, but really should just be using `Time.of`.

Reply:
This logic no longer exists.